dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasableResourceInterface.java

```diff
@@ -56,32 +60,84 @@
         // Gather up any aliases that contain invalid prefixes
         List<String> invalidAliases = new ArrayList<>();
-        if (!user.isCurator() && !user.getIsAdmin()) {
-            invalidAliases = aliases.stream().filter(alias -> StringUtils.startsWithAny(alias, invalidPrefixes)).collect(Collectors.toList());
-        }
+        invalidAliases = aliases.stream().filter(alias -> StringUtils.startsWithAny(alias, INVALID_PREFIXES)).collect(Collectors.toList());
 
         // If there are any aliases with invalid prefixes then report it to the user
         if (invalidAliases.size() > 0) {
             String invalidAliasesString = String.join(", ", invalidAliases);
-            String invalidPrefixesString = String.join(", ", invalidPrefixes);
-            throw new CustomWebApplicationException("These aliases: " + invalidAliasesString + " start with a reserved string,"
+            String invalidPrefixesString = String.join(", ", INVALID_PREFIXES);
+            throw new CustomWebApplicationException("These aliases: " + invalidAliasesString + " start with a reserved string."
                     + " They cannot be used. Please create aliases without these prefixes: " + invalidPrefixesString,
                     HttpStatus.SC_BAD_REQUEST);
         }
+
+        if (blockAliasesWithZenodoFormat) {
+            Pattern pattern = Pattern.compile(ZENDO_DOI_REGEX);
+            List<String> aliasesWithForbiddenFormat = aliases.stream().filter(alias -> pattern.matcher(alias).matches()).collect(Collectors.toList());
+            // If there are any aliases with invalid formats then report it to the user
+            if (aliasesWithForbiddenFormat.size() > 0) {
+                String invalidAliasesString = String.join(", ", aliasesWithForbiddenFormat);
+                throw new CustomWebApplicationException("These aliases : " + invalidAliasesString + " have a format that is forbidden."
+                        + " They cannot be used. Please create aliases without this format: " + ZENDO_DOI_REGEX,
+                        HttpStatus.SC_BAD_REQUEST);
+            }
+        }
     }
 
+    /**
+     * Add aliases to an Entry (e.g. Workflow or Tool)
+     * and check that they are valid before adding them
+     * @param user user authenticated to issue a DOI for the workflow
+     * @param id the id of the Entry
+     * @param aliases a comma separated string of aliases
+     * @return the alias as a string
+     */
     default T addAliases(User user, Long id, String aliases) {
+        return addAliasesAndCheck(user, id, aliases, true);
+    }
+
+    /**
+     * Add aliases to an Entry (e.g. Workflow or Tool)
+     * and check that they are valid before adding them
+     * @param user user authenticated to issue a DOI for the workflow
+     * @param id the id of the Entry
+     * @param aliases a comma separated string of aliases
+     * @return the alias as a string
```

Comment:
JavaDoc for blockFormat missing

Reply:
I tried to add it but the IDE paints it red...